omnigibson/utils/python_utils.py

```diff
@@ -291,6 +291,42 @@ def get_uuid(name, n_digits=8):
     return abs(hash(name)) % (10**n_digits)
 
 
+class StringIntegerMapper:
+    def __init__(self):
+        self._string_to_int = {}
+        self._int_to_string = {}
+        self.next_integer = 0
+
+    def string_to_int(self, s):
+        """
+        Adds a string to the mapping system if not already added and returns the assigned integer.
+
+        Args:
+            s (str): The string to map to an integer.
+
+        Returns:
+            int: The integer assigned to the string.
+        """
+        if s not in self._string_to_int:
```

Comment:
nit: can't we just use len(self._string_to_int) instead of explicitly storing and incrementing self.next_integer?